config/schemas/config.yaml

```diff
@@ -126,6 +363,658 @@ properties:
           - containerd
           - docker
     additionalProperties: false
+  calicoAccountant:
+    additionalProperties: false
+    properties:
+      backend:
+        default: iptables
+        enum:
+          - iptables
+          - nftables
+        type: string
+      enabled:
+        default: true
+        type: boolean
+      resources:
+        $ref: '#/$defs/resources'
+      tolerations:
+        $ref: '#/$defs/tolerations'
+    type: object
+  calicoFelixMetrics:
+    additionalProperties: false
+    properties:
+      enabled:
+        default: true
+        type: boolean
+    type: object
+  certmanager:
+    additionalProperties: false
+    properties:
+      affinity:
+        $ref: '#/$defs/kubernetesAffinity'
+      cainjector:
+        $ref: '#/$defs/component'
+      extraArgs:
+        $ref: '#/$defs/extraArgs'
+      nodeSelector:
+        $ref: '#/$defs/nodeSelector'
+      resources:
+        $ref: '#/$defs/resources'
+      tolerations:
+        $ref: '#/$defs/tolerations'
+      topologySpreadConstraints:
+        $ref: '#/$defs/topologySpreadConstraints'
+      webhook:
+        $ref: '#/$defs/component'
+    type: object
+  ingressNginx:
+    additionalProperties: false
+    properties:
+      controller:
+        additionalProperties: false
+        properties:
+          additionalConfig:
+            type: object
+          affinity:
+            $ref: '#/$defs/kubernetesAffinity'
+          allowSnippetAnnotations:
+            default: false
+            type: boolean
+          chroot:
+            type: boolean
+          config:
+            additionalProperties: false
+            properties:
+              useProxyProtocol:
+                type: boolean
+              annotationsRiskLevel:
+                type: string
+                description: See [reference](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#annotations-risk-level)
+                enum:
+                  - Critical
+                  - High
+                  - Medium
+                  - Low
+            type: object
+          enablepublishService:
+            default: false
+            type: boolean
+          enableAnnotationValidations:
+            description: |-
+              Enable validation of Nginx annotations on Ingress resources.
+              This is kept false by default due to the maturity of the feature,
+              e.g. lack of documentation and disruptive bugs.
+            default: false
+            type: boolean
+          extraArgs: true
+          nodeSelector:
+            $ref: '#/$defs/nodeSelector'
+          resources:
+            $ref: '#/$defs/resources'
+          service:
+            additionalProperties: false
+            properties:
+              annotations:
+                additionalProperties:
+                  type: string
```

Comment:
I tested it on a different cloud provider today and got this error:
```console
wc-config.yaml: ingressNginx.controller.service.annotations.loadbalancer.openstack.org/proxy-protocol: Invalid type. Expected: string, given: boolean
```
This comes from config generated by an internal repository which did not have quotes around the annotation value `true`, and I am not sure if we want to leave it as string to better match the [Kubernetes documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#attaching-metadata-to-objects) which clearly states that keys and values of annotation mappings must be **strings**. Maybe we should fix so we always put quotes around the value of an annotation in our configs, so create a follow up issue(s)? But I wanted to hear others opinions on this @Zash @aarnq

Reply:
Yes, annotations here should follow Kubernetes annotations.

But I don't know if it would be more friendly to enforce that through the template process instead and allow any scalars to be more forgiving. 

(Though this is something that would need to be noted in the description of this parameter.)

Reply:
I observe that there are many annotations throughout that should probably be refactored into a common definition under `$defs`. Future task perhaps?